security/pfSense-pkg-acme/files/usr/local/www/acme/acme_certificates_edit.php

```diff
@@ -0,0 +1,432 @@
+	}
+
+	if($certificate['name'] != "") {
+		$changedesc .= " modified pool: '{$certificate['name']}'";
+	}
+	$certificate['a_domainlist']['item'] = $a_domains;
+	$certificate['a_actionlist']['item'] = $a_actions;
+
+	global $simplefields;
+	foreach($simplefields as $stat) {
+		update_if_changed($stat, $certificate[$stat], $_POST[$stat]);
+	}
+	if (isset($id) && $a_certificates[$id]) {
+		$a_certificates[$id] = $certificate;
+	} else {
+		$a_certificates[] = $certificate;
+	}
+	if (!isset($input_errors)) {
+		if ($changecount > 0) {
+			touch($d_acmeconfdirty_path);
+			write_config($changedesc);
+		}
+		header("Location: acme_certificates.php");
+		exit;
+	}
+}
+
+$closehead = false;
+$pgtitle = array("Services", "Acme", "Certificate options: Edit");
+include("head.inc");
+display_top_tabs_active($acme_tab_array['acme'], "certificates");
+
+// 'processing' done, make all simple fields usable in html.
+foreach($simplefields as $field){
+	$pconfig[$field] = htmlspecialchars($pconfig[$field]);
+}
+
+?>
+<script type="text/javascript">
+	function clearcombo(){
+	  for (var i=document.iform.serversSelect.options.length-1; i>=0; i--){
+		document.iform.serversSelect.options[i] = null;
+	  }
+	  document.iform.serversSelect.selectedIndex = -1;
+	}
+
+	function setCSSdisplay(cssID, display)
+	{
+		var ss = document.styleSheets;
+		for (var i=0; i<ss.length; i++) {
+			var rules = ss[i].cssRules || ss[i].rules;
+			for (var j=0; j<rules.length; j++) {
+				if (rules[j].selectorText === cssID) {
+					rules[j].style.display = display ? "" : "none";
+				}
+			}
+		}
+	}
+	function toggleCSSdisplay(cssID)
+	{
+		var ss = document.styleSheets;
+		for (var i=0; i<ss.length; i++) {
+			var rules = ss[i].cssRules || ss[i].rules;
+			for (var j=0; j<rules.length; j++) {
+				if (rules[j].selectorText === cssID) {
+					rules[j].style.display = rules[j].style.display === "none" ? "" : "none";
+				}
+			}
+		}
+	}
+	
+	function updatevisibility()
+	{
+		d = document;
+		// IE needs components found into javascript variables
+	}
+</script>
+</head>
+<?php
+if (isset($input_errors)) {
+	print_input_errors($input_errors);
+}
+
+$counter=0;
+
+$form = new \Form;
+
+$section = new \Form_Section('Edit Certificate options');
+$section->addInput(new \Form_Input('name', 'Name', 'text', $pconfig['name']
+))->setHelp('The name set here will also be used to create or overwrite a certificate that might already exist with this name in the pfSense certificate m.');
+$section->addInput(new \Form_Input('desc', 'Description', 'text', $pconfig['desc']));
+$activedisable = array();
+$activedisable['active'] = "Active";
+$activedisable['disabled'] = "Disabled";
+$section->addInput(new \Form_Select(
+	'status',
+	'Status',
+	$pconfig['status'],
+	$activedisable
+));
+$a_accountkeys = &$config['installedpackages']['acme']['accountkeys']['item'];
+$section->addInput(new \Form_Select(
+	'acmeaccount',
+	'Acme Account',
+	$pconfig['acmeaccount'],
+	form_name_array($a_accountkeys)
+));
```

Comment:
It would be helpful if this page would warn the user when there are no account keys, and direct them to create an account key. Someone filling out this form could get this far in before realizing it's impossible to complete the page.